src/box/lua/misc.cc

```diff
@@ -215,7 +243,13 @@ lbox_select(lua_State *L)
 	 */
 	port_dump_lua(&port, L, false);
 	port_destroy(&port);
-	return 1; /* lua table with tuples */
+	int ret_count = 1;
+	if (fetch_pos && pos != NULL) {
+		lua_pushlstring(L, pos, pos_end - pos);
+		ret_count++;
+	}
+	region_truncate(&fiber()->gc, svp);
```

Comment:
You should clean up the region even if `box_select` above fails.

Reply:
Done, also changed `lbox_normalize_position` - now it returns int instead of void to clean up the region if it fails.